File: VB/Q338316/CustomReportPreviewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes
Imports DevExpress.Xpf.Printing
Imports DevExpress.XtraPrinting

Namespace Q338316
	Public Class CustomReportPreviewModel
		Inherits ReportPreviewModel
		Private Const XpsExportFormat As String = "Xps"

		Protected Overrides Function CanExport(ByVal parameter As Object) As Boolean
			If CStr(parameter) = XpsExportFormat Then
				Return False
			End If
			Return MyBase.CanExport(parameter)
		End Function

		Protected Overrides Function CanExportToWindow(ByVal parameter As Object) As Boolean
			If CStr(parameter) = XpsExportFormat Then
				Return False
			End If
			Return MyBase.CanExportToWindow(parameter)
		End Function
	End Class
End Namespace

File: CS/E3470/CustomReportPreviewModel.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using DevExpress.Xpf.Printing;
using DevExpress.XtraPrinting;

namespace E3470 {
    public class CustomReportPreviewModel : ReportPreviewModel {
        const string XpsExportFormat = "Xps";

        protected override bool CanExport(object parameter) {
            if((string)parameter == XpsExportFormat)
                return false;
            return base.CanExport(parameter);
        }

        protected override bool CanExportToWindow(object parameter) {
            if((string)parameter == XpsExportFormat)
                return false;
            return base.CanExportToWindow(parameter);
        }
    }
}
